### PR TITLE
update DfmNoteBlockRule.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmNoteBlockRule.cs [hunk 1]
```diff
@@ -7,21 +7,54 @@ namespace Microsoft.DocAsCode.Dfm
...
+                Matcher.CaseInsensitiveString("warning") |
+                Matcher.CaseInsensitiveString("tip") |
+                Matcher.CaseInsensitiveString("important") |
+                Matcher.CaseInsensitiveString("caution")
+            ).ToGroup("notetype") +
+            ']' +
+            Matcher.WhiteSpacesOrEmpty +
+            Matcher.NewLine.RepeatAtLeast(0);
+
         private static readonly Regex _dfmNoteRegex = new Regex(@"^(?<rawmarkdown> *\[\!(?<notetype>(NOTE|WARNING|TIP|IMPORTANT|CAUTION))\] *\n?)(?<text>.*)(?:\n|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromSeconds(10));
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmNoteBlockRule.cs]: >.* [](start = 153, length = 2)

miss this in matcher?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmNoteBlockRule.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmNoteBlockRule.cs]: text group do not use in following code.

**Pull request merged**